core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -19,12 +19,13 @@ trait RTS {
    * error, running forever, or producing an `A`.
    */
   final def unsafeRun[E, A](io: IO[E, A]): A = unsafeRunSync(io) match {
-    case ExitResult.Completed(v)  => v
-    case ExitResult.Terminated(t) => throw t
-    case ExitResult.Failed(e)     => throw Errors.UnhandledError(e)
+    case ExitResult.Completed(v)       => v
+    case ExitResult.Terminated(Nil)    => throw new Error("Fiber interrupted")
```

Comment:
I don't like that, what about adding a new class in `Errors`?

Reply:
Definitely.